controllers/build.go

```diff
@@ -0,0 +1,280 @@
+// PipelineRuns would be that you'd have the git revision appended to the output image tag
+func paramsForWebhookBasedBuilds(component appstudiov1alpha1.Component) []tektonapi.Param {
+	sourceCode := component.Spec.Source.GitSource.URL
+	outputImage := normalizeOutputImageURL(component.Spec.Build.ContainerImage)
+
+	params := []tektonapi.Param{
+		{
+			Name: "git-url",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: sourceCode,
+			},
+		},
+		{
+			Name: "output-image",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: outputImage,
+			},
+		},
+	}
+
+	return params
+}
+
+func commonAnnotations() map[string]string {
+	annotations := map[string]string{
+		"build.appstudio.openshift.io/build":   "true",
+		"build.appstudio.openshift.io/type":    "build",
+		"build.appstudio.openshift.io/version": "0.1",
+	}
+	return annotations
+}
+
+// commonStorage returns the PVC that would be created per namespace for
+// user-triggered and webhook-triggered Tekton workspaces.
+func commonStorage(name string, namespace string) corev1.PersistentVolumeClaim {
+	fsMode := corev1.PersistentVolumeFilesystem
+
+	workspaceStorage := &corev1.PersistentVolumeClaim{
+		ObjectMeta: v1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: commonAnnotations(),
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				"ReadWriteOnce",
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					"storage": resource.MustParse("10Mi"),
+				},
+			},
+			VolumeMode: &fsMode,
+		},
+	}
+	return *workspaceStorage
+}
+
+// route returns the Route resource that would enable
+// ingress traffic into the webhook endpoint ( aka EventListener)
+// TODO: This needs to be secure.
+func route(component appstudiov1alpha1.Component) routev1.Route {
+	var port int32 = 8080
+	webhook := &routev1.Route{
+		ObjectMeta: v1.ObjectMeta{
+			Name:        component.Name,
+			Namespace:   component.Namespace,
+			Annotations: commonAnnotations(),
+		},
+		Spec: routev1.RouteSpec{
+			Path: "/",
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: "el-" + component.Name,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.IntOrString{IntVal: port},
+			},
+		},
+	}
+	return *webhook
+}
+
+// triggerTemplate generates the TriggerTemplate resources
+// which defines how a webhook-based trigger event would be handled -
+// In this case, a PipelineRun to build an image would be created.
+func triggerTemplate(component appstudiov1alpha1.Component) (*triggersapi.TriggerTemplate, error) {
+	webhookBasedBuildTemplate := determineBuildExecution(component, paramsForWebhookBasedBuilds(component), "$(tt.params.git-revision)")
+
+	resoureTemplatePipelineRun := tektonapi.PipelineRun{
+		ObjectMeta: v1.ObjectMeta{
+			GenerateName: component.Name + "-",
```

Comment:
So I tried out your changes, and was able to get a container build working and pushed up to quay.io, which was cool to see.

But, I saw that the pipelinerun had the name `component-sample` (the name of my Component), and stuck around _after_ the build finished or failed, and also stuck around after I deleted the Component and Application CRs.

The other thing I saw was that it was the only pipelinerun that would get created. E.g. if I changed the Component resource to something different, no other pipelineruns would get created. It was like the `GenerateName` field wasn't working for me?

Reply:
That's a known issue I'd say.

The build objects are not being updated upon changes to Components ( unless they are absent/deleted ). I'll be looking into that in the upcoming PRs - especially because the next step is to programmatically persist the tekton resources in git.

Reply:
> The other thing I saw was that it was the only pipelinerun that would get created. E.g. if I updated the Component resource, no other pipelineruns would get created. It was like the GenerateName field wasn't working for me?

I remember why exactly I didn't use the `generateName` for PipelineRun - I was getting too many PipelineRuns since there are usually multiple controller reconciles from the same update sometimes. I'd like to fix that.

> But, I saw that the pipelinerun had the name component-sample (the name of my Component), and stuck around after the build finished or failed, and also stuck around after I deleted the Component and Application CRs.

This is to ensure that the record of the build isn't lost upon deletion of a `Component`. Cleaning up of PipelineRuns would be taken care of a parallel pruning activity ( which isn't in place yet ).

Reply:
I take that back, it makes sense to delete PipelineRuns too upon deletion of a Component, that is the only way to clean the namespace up after usage. 

Let me fix that.

Reply:
If a Component is updated,
* a new PipelineRun is created ( it uses a generateName now )
* the existing TriggerTemplate is updated